src/shared/Core/Interop/MacOS/MacOSEnvironment.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using GitCredentialManager.Interop.Posix;
+
+namespace GitCredentialManager.Interop.MacOS
+{
+    public class MacOSEnvironment : PosixEnvironment
+    {
+        public MacOSEnvironment(IFileSystem fileSystem)
+            : this(fileSystem, GetCurrentVariables()) { }
+
+        internal MacOSEnvironment(IFileSystem fileSystem, IReadOnlyDictionary<string, string> variables)
+            : base(fileSystem)
+        {
+            EnsureArgument.NotNull(variables, nameof(variables));
+            Variables = variables;
+        }
+
+        public override bool TryLocateExecutable(string program, out string path)
+        {
+            string homebrewPrefix = Environment.GetEnvironmentVariable("HOMEBREW_PREFIX");
+            List<string> pathsToIgnore = new List<string>();
+
+            if (homebrewPrefix != null)
+            {
+                pathsToIgnore.Add($"{homebrewPrefix}/Homebrew/Library/Homebrew/shims/shared/git");
+            }
+
+            return TryLocateExecutable(program, pathsToIgnore, out path);
```

Comment:
Could we possibly cache this ignore list to avoid needing to build it on each invocation? Also we should probably use both the `Variables` member to search for variables and `Path.Combine` to combine the paths.

Unit tests initialise the envars field directly but `Environment.GetEnvironmentVariable()` will look at the _real_ environment.

```suggestion
            if (_pathsToIgnore is null)
            {
                _pathsToIgnore = new List<string>();

                if (Variables.TryGetValue("HOMEBREW_PREFIX", out string homebrewPrefix))
                {
                    string homebrewGit = Path.Combine(homebrewPrefix, "Homebrew/Library/Homebrew/shims/shared/git");
                    _pathsToIgnore.Add(homebrewGit);
                }
            }

            return TryLocateExecutable(program, _pathsToIgnore, out path);
```


..where `_pathsToIgnore` is a private `ICollection<string>` field.